x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_migration_client.ts

```diff
@@ -74,4 +98,101 @@ export class RuleMigrationsDataMigrationClient extends RuleMigrationsDataBaseCli
 
     return [migrationDeleteOperation];
   }
+
+  /**
+   * Deletes a migration document by its id.
+   */
+  async saveAsAborted({ id }: { id: string }): Promise<void> {
+    this.logger.info(`Saving migration ${id} as aborted`);
+    const index = await this.getIndexName();
+
+    await this.updateLastExecution({
+      id,
+      lastExecutionParams: {
+        is_aborted: true,
+        error: null,
+        ended_at: new Date().toISOString(),
+      },
+    });
+  }
+  /**
+   * Updates the last execution parameters for a migration document.
+   */
+  private async updateLastExecution({
+    id,
+    lastExecutionParams,
+  }: {
+    id: string;
+    lastExecutionParams: RuleMigrationLastExecution;
+  }): Promise<void> {
```

Comment:
super NIT: 
Move it to the bottom of the class so that all public _saveAs*_ methods are together before this private one.

And since all the parameters are required, maybe it's easier to do:
```suggestion
  private async updateLastExecution(id: string, lastExecution: RuleMigrationLastExecution): Promise<void> {
```